Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.mm

```diff
@@ -92,17 +92,30 @@
 DocumentMap LevelDbRemoteDocumentCache::GetMatching(FSTQuery* query) {
   DocumentMap results;
 
+  // Use the query path as a prefix for testing if a document matches the query.
+  const model::ResourcePath& query_path = query.path;
+  unsigned long immediate_children_path_length = query_path.size() + 1;
```

Comment:
Very optional: `size_t` is probably slightly more appropriate than `unsigned long` (`auto` would also work, though some people dislike `auto` on fundamental types).